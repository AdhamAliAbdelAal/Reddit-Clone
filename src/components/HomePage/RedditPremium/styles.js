import { styled } from '@mui/system';

export const PremiumReddit = styled('div')(() => ({
  padding: 12,
  display: 'flex',
  flexDirection: 'column',
  backgroundColor: '#fff',
  marginTop: 16,
  borderRadius: 7,
  border: '1px solid #ccc',
}));

export const Header = styled('div')(() => ({
  fontSize: 13,
  fontWeight: '500',
}));

export const Paragraph = styled('div')(() => ({
  fontSize: 12,
  fontWeight: '400',
}));
