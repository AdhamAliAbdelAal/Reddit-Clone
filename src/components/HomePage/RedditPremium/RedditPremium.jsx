import { Box } from '@mui/system';
import SportsScoreIcon from '@mui/icons-material/SportsScore';
import RedditButton from '../../RedditButton/RedditButton';
import { Header, Paragraph, PremiumReddit } from './styles';
/**
 * This component contains info about reddit premium.
 *
 * @component RedditPremium
 * @returns {React.Component}  Information about Premium version of reddit.
 */

function RedditPremium() {
  return (
    <PremiumReddit>
      <Box display="flex" mb={2}>
        <SportsScoreIcon
          sx={{
            fontSize: 40,
            mx: 1,
          }}
          color="secondary"
        />
        <Box display="flex" flexDirection="column">
          <Header>
            Reddit Premium
          </Header>
          <Paragraph>
            The best Reddit experience, with monthly Coins
          </Paragraph>
        </Box>
      </Box>
      <RedditButton variant="contained" color="secondary" padding="4px" fontSize={15} fontWeight="bold">
        try now
      </RedditButton>
    </PremiumReddit>
  );
}

export default RedditPremium;
