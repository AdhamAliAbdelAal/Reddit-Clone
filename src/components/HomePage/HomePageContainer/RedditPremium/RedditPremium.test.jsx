import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import RedditPremium from './RedditPremium';

describe(RedditPremium, () => {
  it('RedditPremium renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <RedditPremium />
          ,
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
