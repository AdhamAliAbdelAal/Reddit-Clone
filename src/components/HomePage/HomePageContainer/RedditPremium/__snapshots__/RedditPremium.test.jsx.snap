// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedditPremium RedditPremium renders correctly 1`] = `
Array [
  <div
    className="css-170bi6"
  >
    <div
      className="MuiBox-root css-1kukc4r"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-zvbc5x-MuiSvgIcon-root"
        data-testid="SportsScoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 6H9V4h2v2zm4-2h-2v2h2V4zM9 14h2v-2H9v2zm10-4V8h-2v2h2zm0 4v-2h-2v2h2zm-6 0h2v-2h-2v2zm6-10h-2v2h2V4zm-6 4V6h-2v2h2zm-6 2V8h2V6H7V4H5v16h2v-8h2v-2H7zm8 2h2v-2h-2v2zm-4-2v2h2v-2h-2zM9 8v2h2V8H9zm4 2h2V8h-2v2zm2-4v2h2V6h-2z"
        />
      </svg>
      <div
        className="MuiBox-root css-j7qwjs"
      >
        <div
          className="css-1tpqrfm"
        >
          Reddit Premium
        </div>
        <div
          className="css-1123vha"
        >
          The best Reddit experience, with monthly Coins
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1a8l933-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      fontSize={15}
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="4px"
      tabIndex={0}
      type="button"
    >
      try now
    </button>
  </div>,
  ",",
]
`;
